### PR TITLE
Fixes #12603: Wrong reporting in "ManagekeyValue" technique

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/12603

### CODE REVIEW

File: techniques/fileDistribution/manageKeyValueFile/1.3/manage-key-value-file.st [hunk 1]
```diff
@@ -36,10 +36,13 @@ bundle agent manage_key_value_file_RudderUniqueID {
...
 
-    "canonified_file[${index}]"      string => canonify("${file[${index}]}");
+    "args"  slist => { "${file[${index}]}", "${key[${index}]}", "${value[${index}]}","${separator[${index}]}", "${option[${index}]}" };
+    "report_param[${index}]"         string => join("_", args);
+    "canonified_params[${index}]"    string => canonify("${report_param[${index}]}");
     "canonified_hook[${index}]"      string => canonify("${post_hook[${index}]}");
-    "class_prefix_${index}"          string => "file_ensure_key_value_${canonified_file[${index}]}";
-    "hook_class_prefix_${index}"      string => "command_execution_${canonified_hook[${index}]}";
+
+    "class_prefix_${index}"          string => "file_ensure_key_value_${canonified_params[${index}]}";
```
Comment (amousset) [inline techniques/fileDistribution/manageKeyValueFile/1.3/manage-key-value-file.st]: I think this will only work with 4.3 and the new class prefix

**Comment by Fdall** [comment]
Rebasing the PR, the problem only appears in rudder 4.3 and later.

**Comment by Fdall** [comment]
PR rebased
4.3

**Comment by Fdall** [comment]
PR replaced by https://github.com/Normation/rudder-techniques/pull/1299

**Pull request closed without merging**